File: admin/src/main/java/com/imstargg/admin/domain/BattleEvent.java
package com.imstargg.admin.domain;

import com.imstargg.storage.db.core.brawlstars.BattleEventCollectionEntity;
import jakarta.annotation.Nullable;

import java.time.OffsetDateTime;

public record BattleEvent(
        BattleEventCollectionEntity entity,
        BattleEventMap map,
        @Nullable String battleMode,
        @Nullable OffsetDateTime latestBattleTime,
        boolean soloRanked
) {
}

File: admin/imstargg-admin-frontend/src/model/BattleEvent.ts
import {BattleEventMap} from "@/model/BattleEventMap";
import {BattleEventEntity} from "@/model/entity/BattleEventEntity";

export interface BattleEvent {
  entity: BattleEventEntity;
  map: BattleEventMap;
  battleMode: string | null;
  latestBattleTime: Date | null;
  soloRanked: boolean;
}